kvstore/badger/badger.go

```diff
@@ -1,54 +1,43 @@
-package smt
+package badger
 
 import (
 	"errors"
 	"fmt"
 	"io"
 
-	badger "github.com/dgraph-io/badger/v4"
+	"github.com/dgraph-io/badger/v4"
+	"github.com/pokt-network/smt/kvstore"
 )
 
-// KVStore is an interface that defines a key-value store
-// that can be used standalone or as the node store for an SMT.
-type KVStore interface {
-	// Store methods
-	Get(key []byte) ([]byte, error)
-	Set(key, value []byte) error
-	Delete(key []byte) error
-
-	// Lifecycle methods
-	Stop() error
-
-	// Data methods
-	Backup(writer io.Writer, incremental bool) error
-	Restore(io.Reader) error
-
-	// Accessors
-	GetAll(prefixKey []byte, descending bool) (keys, values [][]byte, err error)
-	Exists(key []byte) (bool, error)
-	ClearAll() error
-	Len() int
-}
-
 const (
 	maxPendingWrites = 16 // used in backup restoration
 )
 
-var _ KVStore = &badgerKVStore{}
-
-var (
-	ErrKVStoreExists    = errors.New("kvstore already exists")
-	ErrKVStoreNotExists = errors.New("kvstore does not exist")
-)
+var _ kvstore.KVStore = &badgerKVStore{}
 
 type badgerKVStore struct {
 	db          *badger.DB
 	last_backup uint64 // timestamp of the most recent backup
 }
 
+var badgerToSMTErrorsMap = map[error]error{
```

Comment:
Can we add some quick comments explaining this map and the conversion function in the godoc format